src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -0,0 +1,28 @@
+import type { QueryRawOptions, Sequelize } from '../../sequelize';
+import type { AbstractQueryGenerator } from './query-generator';
+
+export interface QueryInterfaceOptions {
+  sequelize: Sequelize;
+  queryGenerator: AbstractQueryGenerator;
+}
+
+export class AbstractQueryInterfaceTypeScript {
+  readonly sequelize: Sequelize;
+  readonly queryGenerator: AbstractQueryGenerator;
+
+  constructor(options: QueryInterfaceOptions) {
+    this.sequelize = options.sequelize;
+    this.queryGenerator = options.queryGenerator;
+  }
+
+  /**
+   * Queries the schema (table list).
+   *
+   * @param schema The schema to query. Applies only to Postgres.
+   * @param options
+   */
+  async createSchema(schema: string, options?: QueryRawOptions): Promise<void> {
+    const sql = this.queryGenerator.createSchemaQuery(schema);
```

Comment:
Probably out of scope for this PR but;
We have `rejectInvalidOptions` implemented for `createSchemaQuery`. Why don't we pick the options that are in `CreateSchemaQueryOptions` and pass them on? (so not all options which are part of `QueryRawOptions` but only `collate` and `charset` for now)